### DISCUSSION TITLE
Resize windows when dock size changes

### DISCUSSION THREAD

**Discussion by mcrapts** [rxhanson/RectanglePro-Community#581] [Ideas]
I have the dock positioned on the left side of my screen. The dock size can change depending on the number of apps, so if I open or close apps the dock can grow or shrink. If windows are moved to the left half of the screen, this can result in a gap between the dock and the window if the dock shrinks, or the window can move slightly behind the dock if it grows. This is especially the case when connecting or disconnecting external displays. 

Ideally Rectangle would be able to recognize the changing dock size and reposition the windows accordingly when this happens. The newly introduced window tiling feature in macOS Sequoia does this, which is really nice, but unfortunately it lacks a lot of features compared to Rectangle so it would be great if Rectangle Pro could support this as well.

**Comment by Aaron-Rumpler** [comment]
This is something I've been considering requesting for a while as well. It doesn't just affect users with the dock on the left/right of the screen, I see it too with my 14" MBP with the dock at the bottom and set to the default size.

**Comment by rxhanson** [comment]
As far as I know, as a 3rd party this would require polling. Would you still be interested in this if Rectangle/Pro were using CPU cycles polling in the background for this?

**Reply by Aaron-Rumpler** [reply]
Maybe make it optional in that case, but I definitely would be. I can't imagine it would be much compared to all the other stuff running in the background?

**Reply by mcrapts** [reply]
I guess it depends on the impact on CPU usage. If it's negligible and optional I think it's worth it.
